### PR TITLE
Enhancement: Replace <img> tags with Next.js Image Component #824

### PR DESCRIPTION
**Pull request by 29deepanshutyagi**
Replaced all img HTML tags with the Image component across the codebase for better optimization and consistency.
Removed certain unnecessary configurations and added the error level to the ESLint configuration to enforce stricter code quality.
Ensured the changes pass linting and the build process before submission.
Removed outdated drafts and unnecessary files to streamline the project structure.

Changes are pushed through the img branch.

### CODE REVIEW

**Comment by DarhkVoyd** [comment]
Hey could you please link the related issue number in your description.

**Comment by 29deepanshutyagi** [comment]
issue #824

File: .eslintrc.js [hunk 1]
```diff
@@ -53,7 +56,8 @@ module.exports = {
     'keyword-spacing': ['error', { before: true, after: true }],
     'comma-spacing': ['error', { before: false, after: true }],
     'key-spacing': ['error', { beforeColon: false, afterColon: true }],
-    '@next/next/no-img-element': 'off',
+    // Change this rule from 'off' to 'error'
```
Comment (DarhkVoyd) [inline .eslintrc.js:59]: What is the need of this comment? This PR does change it to 'error'.

File: components/GettingStarted.tsx
Comment (DarhkVoyd) [inline components/GettingStarted.tsx:1]: I do not understand the changes made to this component. Could you please explain how is this relevant to this PR?

File: components/JsonEditor.tsx [hunk 2]
```diff
@@ -294,24 +295,31 @@ export default function JsonEditor({ initialCode }: { initialCode: string }) {
...
             data-test='check-json-schema'
           >
             {isJsonSchema ? (
               <>
-                <img src='/logo-white.svg' className='h-4 mr-1.5' /> schema
+                <Image
+                  src='/logo-white.svg'
+                  alt=' logo-white'
+                  className='h-4 mr-1.5'
+                />{' '}
```
Comment (DarhkVoyd) [inline components/JsonEditor.tsx:321]: I believe `{' '}` is a unwanted spacing added by your formatter.

File: components/Layout.tsx [hunk 3]
```diff
@@ -10,6 +10,8 @@ import { useTheme } from 'next-themes';
 import DarkModeToggle from './DarkModeToggle';
 import extractPathWithoutFragment from '~/lib/extractPathWithoutFragment';
 import ScrollButton from './ScrollButton';
+import Image from 'next/image';
+import { title } from 'process';
```
Comment (DarhkVoyd) [inline components/Layout.tsx:14]: Why are we importing `process.title`, seems like a mistake?

File: components/Layout.tsx
Comment (DarhkVoyd) [inline components/Layout.tsx:1]: We probably don't need much of these simple comments.

File: components/Sidebar.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { SegmentHeadline } from './Layout';
 import extractPathWithoutFragment from '~/lib/extractPathWithoutFragment';
 import CarbonAds from './CarbonsAds';
 import { useTheme } from 'next-themes';
-import ExternalLinkIcon from '../public/icons/external-link-black.svg';
```
Comment (DarhkVoyd) [inline components/Sidebar.tsx:11]: Why was this removed?

File: pages/index.page.tsx [hunk 5]
```diff
@@ -206,7 +207,7 @@ const Home = (props: any) => {
   const [airbnb_logo, setAirbnb_logo] = useState('');
   const [postman_logo, setPostman_logo] = useState('');
   const [itflashcards_logo, setItflashcards_logo] = useState('');
-  const [route4me_logo, setRoute4me_logo] = useState('');
```
Comment (DarhkVoyd) [inline pages/index.page.tsx:209]: Why do we no longer need this?

**Review by DarhkVoyd — CHANGES_REQUESTED** [review]
Hey! thanks for working on this issue. 
1. I just left a few comments on some individual changes, please consider those. 
2. More importantly, many of these changes will actually cause errors on pages, not limited to `/docs` and `/tools` which are two important pages, due the the fact that the `next/image` component expects `width` and `height` attributes which is missing from most of the instances.
Note: The errors are suppressed in the production build so you will have to check/debug them in development.

File: pages/index.page.tsx (hunk 5 above)
Comment (29deepanshutyagi) [reply pages/index.page.tsx:209]: sure , i will resolve these issues and will raise pr soon

Comment (DarhkVoyd) [reply pages/index.page.tsx:209]: You don't have to close this PR, you can continue adding commits to this.
